components/mkfatfs/src/idf/modified/driver/include/driver/sdspi_host.h

```diff
@@ -0,0 +1,156 @@
+// Copyright 2015-2017 Espressif Systems (Shanghai) PTE LTD
```

Comment:
can we use mkfatfs via a submodule as well?

Reply:
I would love to.  I looked at doing that but couldn't figure out how.
The problem is that the source directory for that project is several layers deep.  
https://github.com/jkearins/ESP32_mkfatfs/tree/master/components/mkfatfs/src

And it is my understanding that you can't check-out a sub-directory of a repo as a submodule.

Ideas?

Reply:
@phatpaul thoughts about asking jkearins to split that off into a separate repository so other people can reuse it? You are correct that git doesn't allow what we need because he embedded his component into the subdirectory. There are some hacks to make it work like add a submodule to another directory like repos/ and then symlink from repos/mkfatfs/components/mkfatfs/src to components/mkfatfs but that seems kind of blah.